Game/Scripts/BossShieldAttackScript.cpp

```diff
@@ -0,0 +1,252 @@
+#include "StdAfx.h"
+#include "BossShieldAttackScript.h"
+
+//#include "Application.h"
+//#include "Modules/ModuleScene.h"
+//#include "Scene/Scene.h"
+
+#include "Components/ComponentScript.h"
+#include "Components/ComponentTransform.h"
+#include "Components/ComponentRigidBody.h"
+
+#include "../Scripts/EnemyClass.h"
+#include "../Scripts/BossShieldScript.h"
+#include "../Scripts/EnemyDroneScript.h"
+#include "../Scripts/EnemyVenomiteScript.h"
+#include "../Scripts/HealthSystem.h"
+
+REGISTERCLASS(BossShieldAttackScript);
+
+BossShieldAttackScript::BossShieldAttackScript() : Script(), bossShieldObject(nullptr), isShielding(false),
+	shieldingTime(0.0f), shieldingMaxTime(20.0f), triggerShieldAttackCooldown(false), shieldAttackCooldown(0.0f),
+	shieldAttackMaxCooldown(50.0f), triggerEnemySpawning(false), enemiesToSpawnParent(nullptr),
+	enemySpawnTime(0.0f), enemyMaxSpawnTime(2.0f), battleArenaAreaSize(nullptr), healthSystemScript(nullptr)
+{
+	REGISTER_FIELD(shieldingMaxTime, float);
+	REGISTER_FIELD(shieldAttackMaxCooldown, float);
+
+	REGISTER_FIELD(bossShieldObject, BossShieldScript*);
+
+	REGISTER_FIELD(enemyMaxSpawnTime, float);
+
+	REGISTER_FIELD(enemiesToSpawnParent, GameObject*);
+
+	REGISTER_FIELD(battleArenaAreaSize, ComponentRigidBody*);
+}
+
+void BossShieldAttackScript::Start()
+{
+	shieldingTime = shieldingMaxTime;
+
+	if (enemiesToSpawnParent == nullptr)
+	{
+		return;
```

Comment:
```suggestion
		LOG_ERROR("enemiesToSpawnParent not set in BossShieldAttackScript owned by {}", GetOwner());
		throw ComponentNotFoundException();
```
This is just a temporary fix until we merge #654. If the script can run without the code below this check (empty vectors and no health system), no need to throw an exception, we can just like return like we do now.